### PR TITLE
#2994 fix payment summary discount rate

### LINKED ISSUE #2994: Insurance discount rate not displaying correctly in payment summary
## Describe the bug

Insurance discount rate does not display correctly in the payment window.

### To reproduce

1. Create a new prescription for items with a non-zero sell price.
2. Create/select an insurance policy.
3. Note that the insurance discount amount correctly updates, but the rate remains `"0%"`.

### Expected behaviour

Discount rate should update correctly in payment window.

### Proposed Solution

Fix up existing discount rate selectors and add wrapper for payments for consistency with `discountAmount`.

### Version and device info

- App version: v4.0.4.
- Tablet model: Android emulator (Nexus 10).
- OS version: API 28.

### Additional context

N/A.

### PR DESCRIPTION
**Pull request by wlthomson**
Fixes #2994.

## Change summary

Fixes a few errors with insurance rate, including a few redux changes to make payment and insurance logic a bit clearer. 

## Testing

Steps to reproduce or otherwise test the changes of this PR:

- [ ] When selecting an insurance policy in the payment window,  the insurance discount rate updates correctly.
- [ ] After creating a new insurance policy in the payment window,  the insurance discount rate updates correctly.
- [ ] Edit an insurance policy in the payment window (ensure the policy has a discount rate which differs from the default of `25`). Cancel without making any changes. Create a new insurance policy. The insurance slider is reset to the default value of `25` (not the value of the previously edited policy).
- [ ] Create an insurance policy in the payment window. Cancel without creating the policy. Edit an existing policy (ensure the policy has a discount rate which differs from the default of `25`). The insurance slider is set to the correct value (not the default value of `25`).

### Related areas to think about

Depending on turn-around, this will be the last PR for `v5.1.0` or a patch `v5.1.1` (planning to release `v5.1.0` this arvo).

### CODE REVIEW

**Comment by wlthomson** [comment]
Just merged to get into `v5.1.0` for release. Need to do a patch anyway, so feel free to review and I will action as part of that!

**Comment by mark-prins** [comment]
> Just merged to get into `v5.1.0` for release. Need to do a patch anyway, so feel free to review and I will action as part of that!

am still reviewing. Code look ok - having trouble testing after some recent mSupply changes.

File: src/widgets/Slider.js [hunk 1]
```diff
@@ -47,6 +47,13 @@ export const Slider = React.forwardRef(
   ) => {
     if (!onEndEditing) throw new Error('Must provide onEndEditing prop!');
 
+    React.useEffect(() => {
+      if (value !== currentValue) {
+        setString(value);
+        setSlider(value);
+      }
+    }, [value]);
```
Comment (joshxg) [inline src/widgets/Slider.js:55]: weirdo JS hoisting !

File: src/reducers/PaymentReducer.js [hunk 2]
```diff
@@ -15,7 +15,7 @@ const initialState = () => ({
   transaction: null,
   patient: null,
   paymentAmount: currency(0),
-  insurancePolicy: null,
+  discountRate: 0,
```
Comment (joshxg) [inline src/reducers/PaymentReducer.js:18]: much better. I wonder if we can take it even further and not have the redundant `discountRate` at all.. probably not!

**Comment by mark-prins** [comment]
Testing all good. Thanks @wlthomson!!

**Pull request merged**